### DISCUSSION TITLE
`read_json`, Unable to connect to URL. 405 (Method Not Allowed). POST vs GET?

### DISCUSSION THREAD

**Discussion by kfollesdal** [duckdb/duckdb#15120] [Q&A]
I try to load data with `read_json` with the following query (duckdb 1.1.3):
```
SELECT * FROM read_json('https://biapi.nve.no/magasinstatistikk/api/Magasinstatistikk/HentOffentligData');
```
but I get error:
```
IO Error: Unable to connect to URL "https://biapi.nve.no/magasinstatistikk/api/Magasinstatistikk/HentOffentligData": 405 (Method Not Allowed)
```
Is duckdb using `POST` when making the request? For if I use `curl` I get data but if I use `curl -X POST` i get the same error status:
```
curl -X POST -v https://biapi.nve.no/magasinstatistikk/api/Magasinstatistikk/HentOffentligData
...
< HTTP/1.1 405 Method Not Allowed
...
< Allow: GET
...
```
Just for checking, this works as expected:
```
SELECT * FROM read_json('https://api.github.com/repos/duckdb/duckdb');
```

**Comment by Tishj** [comment]
I imagine it's the HEAD request that's failing, or the [range request](https://www.rfc-editor.org/rfc/rfc9110.html#field.range)

**Comment by kfollesdal** [comment]
Yes, `HEAD` request is not allowed for that url:
`curl --head https://biapi.nve.no/magasinstatistikk/api/Magasinstatistikk/HentOffentligData`
gives `405`
```
HTTP/1.1 405 Method Not Allowed
Content-Length: 0
Date: Thu, 05 Dec 2024 09:05:13 GMT
Server: Kestrel
Allow: GET
Set-Cookie: ARRAffinity=ccbd0dc934f1a1384a149c52d5ad3a92aa4445227976bb7dc81a29733c22fc6a;Path=/;HttpOnly;Secure;Domain=biapi.nve.no
Set-Cookie: ARRAffinitySameSite=ccbd0dc934f1a1384a149c52d5ad3a92aa4445227976bb7dc81a29733c22fc6a;Path=/;HttpOnly;SameSite=None;Secure;Domain=biapi.nve.no
X-Powered-By: ASP.NET
```
any way to tell duckdb only to use `GET`?
As you can see from response `Allow: GET`.

**Comment by kfollesdal** [comment]
Maybe this solves the problem? https://github.com/duckdb/duckdb/pull/14979
